src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -5342,7 +5342,6 @@ internal static class PSSnapInHelpers
                 {"Add-History",                       new SessionStateCmdletEntry("Add-History", typeof(AddHistoryCommand), helpFile) },
                 {"Clear-History",                     new SessionStateCmdletEntry("Clear-History", typeof(ClearHistoryCommand), helpFile) },
                 {"Debug-Job",                         new SessionStateCmdletEntry("Debug-Job", typeof(DebugJobCommand), helpFile) },
-#if !UNIX
                 {"Disable-PSRemoting",                new SessionStateCmdletEntry("Disable-PSRemoting", typeof(DisablePSRemotingCommand), helpFile) },
                 {"Enable-PSRemoting",                 new SessionStateCmdletEntry("Enable-PSRemoting", typeof(EnablePSRemotingCommand), helpFile) },
                 {"Get-PSHostProcessInfo",             new SessionStateCmdletEntry("Get-PSHostProcessInfo", typeof(GetPSHostProcessInfoCommand), helpFile) },
```

Comment:
Are these cmdlets actually work on Unix platform today? At lease the `PSHostProcess` related cmdlets are not ...

Reply:
/cc @PaulHigin Could you please help?

Reply:
These cmdlets are left out of UNIX platforms because they only work on Windows platforms (with WinRM).  The exceptions are the *PSHostProcess* cmdlets that haven't yet been ported to UNIX platforms.  But with new support for named pipes from DotNet Standard these can probably be implemented on UNIX fairly easily, and I hope to do so in the not too distant future.

Even though we have the OMI bridge that allows UNIX -> Windows WinRM remoting, and limited Windows WinRM -> UNIX remoting, these cmdlet still won't run correctly on UNIX platforms and so are removed from the build.

Reply:
The assertion should check for a different number quality on Unix platform.

Reply:
Well [the assertion](https://github.com/PowerShell/PowerShell/blob/4e3db1bb93b17228752de9e3e99f89defbdc01b0/src/System.Management.Automation/engine/InitialSessionState.cs#L5113) is checking the [number of cmdlets in the cache](https://github.com/PowerShell/PowerShell/blob/4e3db1bb93b17228752de9e3e99f89defbdc01b0/src/System.Management.Automation/engine/InitialSessionState.cs#L4991) against the [hard-coded cmdlets we expect at startup](https://github.com/PowerShell/PowerShell/blob/4e3db1bb93b17228752de9e3e99f89defbdc01b0/src/System.Management.Automation/engine/InitialSessionState.cs#L5334).

The problem is that on UNIX-like, we're not expecting the remoting cmdlets in the hard-coded list, but they are present in the cache for some reason. So the question is, if we don't support those cmdlets on UNIX-like, why are they there in the cache and what happens when you try and use one?

The tests seem to be a runtime version of this assertion by the way. A similar hard-coded list where we check cmdlet availability.